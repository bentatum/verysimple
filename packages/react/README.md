# @verysimple/react

This is the `@verysimple/react` package