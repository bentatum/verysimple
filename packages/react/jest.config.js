/** @type {import('jest').Config} */
module.exports = {
  testEnvironment: "jsdom",
  setupFilesAfterEnv: ["@testing-library/jest-dom/extend-expect"],
  collectCoverage: true,
  coverageThreshold: {
    global: {
      statements: 75,
      branches: 72,
      functions: 75,
      lines: 75,
    },
  },
  coverageReporters: ["text-summary", "json-summary"],
  collectCoverageFrom: ["src/**/*.{ts,tsx}", "!**/*.d.ts"],
  coverageDirectory: "coverage",
  moduleNameMapper: {
    "@/test/(.*)": "<rootDir>/test/$1",
    "@/(.*)$": "<rootDir>/src/$1",
  },
  modulePathIgnorePatterns: ["<rootDir>/dist/"],
  prettierPath: null,
};
