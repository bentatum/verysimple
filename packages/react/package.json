{
  "name": "@verysimple/react",
  "version": "8.1.1",
  "description": "A very simple component library",
  "main": "dist/index.js",
  "files": [
    "dist"
  ],
  "author": "Ben Tatum <mail@benjtatum.com>",
  "license": "MIT",
  "repository": "https://github.com/bentatum/verysimple",
  "scripts": {
    "test": "jest",
    "prebuild": "rm -rf dist",
    "build": "tsc --project tsconfig.build.json && tsc-alias"
  },
  "peerDependencies": {
    "react": ">=16",
    "react-dom": ">=16"
  },
  "devDependencies": {
    "@babel/core": "^7.20.12",
    "@babel/preset-env": "^7.19.1",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.18.6",
    "@testing-library/dom": "^8.20.0",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.4.0",
    "@testing-library/user-event": "^14.4.3",
    "@types/inquirer": "latest",
    "@types/jest": "^29.2.3",
    "@types/react": "^18.0.21",
    "@types/testing-library__jest-dom": "^5.14.5",
    "babel-jest": "^29.0.3",
    "babel-loader": "^8.2.5",
    "classnames": "^2.3.2",
    "framer-motion": "^10.15.0",
    "jest": "^29.0.3",
    "jest-environment-jsdom": "^29.0.3",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "tailwindcss": "^3.1.8",
    "tsc-alias": "^1.8.2",
    "typescript": "^4.9.3"
  },
  "volta": {
    "node": "20.10.0",
    "yarn": "1.22.21"
  },
  "dependencies": {
    "@headlessui/react": "^2.1.9",
    "@heroicons/react": "^2.0.13",
    "clsx": "^2.1.0",
    "framer-motion": "^10.15.0"
  }
}
