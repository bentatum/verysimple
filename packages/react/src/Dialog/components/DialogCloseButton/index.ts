export { default } from './DialogCloseButton';
export * from './DialogCloseButton';
