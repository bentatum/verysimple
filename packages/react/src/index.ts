export * from "./ButtonBase";
export * from "./Card";
export * from "./Dialog";
export * from "./FilledButtonBase";
export * from "./FilledIconButton";
export * from "./GradientBorder";
export * from "./helpers";
export * from "./IconButton";
export * from "./NormalIconButton";
export * from "./OutlinedButtonBase";
export * from "./OutlinedIconButton";
export * from "./Spinner";
export * from "./types";
