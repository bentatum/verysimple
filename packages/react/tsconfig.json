{
  "extends": "@verysimple/tsconfig/react-library.json",
  "compilerOptions": {
    "outDir": "./dist",
    "paths": {
      "@/*": ["./src/*"]
    }
  },
  "include": ["src/**/*.ts", "src/**/*.tsx"],
  "exclude": ["node_modules"]
}
