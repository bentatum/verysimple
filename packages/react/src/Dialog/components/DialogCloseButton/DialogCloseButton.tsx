import { IconButton, IconButtonProps } from "@/IconButton";
import { forwardRef } from "react";
import { XMarkIcon } from "@heroicons/react/24/solid";

export interface DialogCloseButtonProps extends IconButtonProps {}

export const DialogCloseButton = forwardRef<
  HTMLButtonElement,
  DialogCloseButtonProps
>((props, ref) => {
  return (
    <IconButton aria-label="Close" {...props} ref={ref}>
      <XMarkIcon className="h-7 w-7" />
    </IconButton>
  );
});

DialogCloseButton.displayName = "DialogCloseButton";

export default DialogCloseButton;
