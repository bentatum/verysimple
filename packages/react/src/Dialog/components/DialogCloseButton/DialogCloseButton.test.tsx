import { render, screen } from "@testing-library/react";
import { createRef } from "react";
import DialogCloseButton from "./DialogCloseButton";

describe("DialogCloseButton", () => {
  test("snapshot", () => {
    const { container } = render(<DialogCloseButton />);
    expect(container).toMatchSnapshot();
  });

  test("render a button", () => {
    render(<DialogCloseButton />);
    expect(screen.getByRole("button", { name: "Close" })).toBeInTheDocument();
  });

  test("icon", () => {
    const { container } = render(<DialogCloseButton />);
    expect(container.querySelector("svg")).toBeInTheDocument();
  });

  it("forwards the ref", () => {
    const ref = createRef<HTMLButtonElement>();
    render(<DialogCloseButton ref={ref} />);
    expect(ref.current).toBeInTheDocument();
  });
  it("has a display name", () => {
    expect(DialogCloseButton.displayName).toBe("DialogCloseButton");
  });
});
