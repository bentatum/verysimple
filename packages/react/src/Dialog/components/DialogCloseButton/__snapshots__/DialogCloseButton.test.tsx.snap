// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogCloseButton snapshot 1`] = `
<div>
  <button
    aria-label="Close"
    class="focus:outline-none cursor-pointer disabled:cursor-not-allowed inline-flex items-center justify-center min-w-[48px] shadow-sm focus:shadow-md hover:shadow-md disabled:shadow-none transition-shadow ease-in duration-100 rounded-2xl text-base min-h-[44px] focus:bg-zinc-300/20 hover:bg-zinc-300/20 active:bg-zinc-300/20 dark:focus:bg-zinc-750/60 dark:hover:bg-zinc-750/60 dark:active:bg-zinc-750/60"
    tabindex="0"
  >
    <svg
      aria-hidden="true"
      class="h-7 w-7"
      data-slot="icon"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M5.47 5.47a.75.75 0 0 1 1.06 0L12 10.94l5.47-5.47a.75.75 0 1 1 1.06 1.06L13.06 12l5.47 5.47a.75.75 0 1 1-1.06 1.06L12 13.06l-5.47 5.47a.75.75 0 0 1-1.06-1.06L10.94 12 5.47 6.53a.75.75 0 0 1 0-1.06Z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;
